src/editor/gridstack/gridstackLayout.ts

```diff
@@ -34,9 +34,45 @@ export class GridStackLayout extends Layout {
     this._cellHeight = info.defaultCellHeight;
     this._columns = info.maxColumns;
 
-    this._gridItems = new Array<GridStackItem>();
+    this._margin = info.cellMargin;
+    this._cellHeight = info.defaultCellHeight;
+    this._columns = info.maxColumns;
 
-    this._gridItemChanged = new Signal<this, GridStackNode[]>(this);
+    this._gridHost = document.createElement('div');
+    this._gridHost.className = 'grid-stack';
+
+    this._grid = GridStack.init(
```

Comment:
Defining the grid in the `constructor` to avoid the potential `undefined` case, which helps reducing the use of `?` and `!`.

Then the node is added to the parent in the layout `init()` method below.